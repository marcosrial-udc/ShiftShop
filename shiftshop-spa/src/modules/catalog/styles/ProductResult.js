import {makeStyles} from '@material-ui/core';

const useStyles = makeStyles(theme => ({
    emptyPlaceholder: {
        padding: theme.spacing(1),
        textAlign: 'center'
    },
    image: {
        maxWidth: '100%',
        [theme.breakpoints.down('sm')]: {maxHeight: '20rem'},
        [theme.breakpoints.up('md')]: {maxHeight: '25rem',},
    },
    emptyText: {
        fontStyle: 'italic',
        fontWeight: 500
    },
    backToProducts: {marginTop: theme.spacing(2)}
}));

export default useStyles;
