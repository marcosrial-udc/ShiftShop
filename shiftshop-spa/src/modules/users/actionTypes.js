export const GET_ROLES_COMPLETED = "project/users/getRolesCompleted";
export const LOGIN_COMPLETED = "project/users/loginCompleted";
export const LOGOUT = "project/users/logout";
export const GET_USERS_COMPLETED = "project/users/getUsersCompleted";
export const CLEAR_USERS_SEARCH = "project/users/clearUsersSearch";

