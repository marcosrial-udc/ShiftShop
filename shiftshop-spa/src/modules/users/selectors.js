const getModuleState = state => state.users;

export const getUser = state =>
    getModuleState(state).user;

export const isLoggedIn = state =>
    getUser(state) !== null;
