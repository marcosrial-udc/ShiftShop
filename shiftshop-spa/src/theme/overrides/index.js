import MuiTableCell from './MuiTableCell';
import MuiTableHead from './MuiTableHead';
import MuiTextField from './MuiTextField';

export default {
    MuiTableCell,
    MuiTableHead,
    MuiTextField,
};
