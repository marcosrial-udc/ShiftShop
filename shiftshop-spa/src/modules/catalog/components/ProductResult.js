import React, {Fragment, useState} from 'react';
import {useSelector} from 'react-redux';
import {useHistory} from 'react-router-dom';
import PropTypes from 'prop-types';
import {FormattedMessage} from 'react-intl';
import {Button, Card, CardActions, CardContent, CardHeader, Divider, IconButton, Typography} from '@material-ui/core';
import {Edit} from '@material-ui/icons';

import {notFound} from '../../../assets/images';
import useStyles from '../styles/ProductResult';

import {formulas} from '../../../utils'
import users, {Role} from '../../users';
import ProductDetails from './ProductDetails';
import EditProduct from './EditProduct';
import ProductProfitText from './ProductProfitText';

const ProductResult = ({product}) => {
    const classes = useStyles();

    const user = useSelector(users.selectors.getUser);
    const hasRole = roles => users.selectors.hasRole(user, roles);

    const [open, setOpen] = useState(false);

    const history = useHistory();

    if (!product) {
        return (
            <div className={classes.emptyPlaceholder}>
                <img className={classes.image} src={notFound} alt="No product"/>
                <Typography className={classes.emptyText} variant="subtitle1">
                    <FormattedMessage id="project.catalog.ProductResult.notFound"/>
                </Typography>
                <Button className={classes.backToProducts} variant="outlined"
                        onClick={() => history.push('/catalog/products')}>
                    <FormattedMessage id="project.catalog.ProductResult.backToProducts"/>
                </Button>
            </div>
        );
    }

    return (
        <Card>
            <CardHeader title={<FormattedMessage id="project.catalog.Product.title"/>}
                        action={hasRole([Role.ADMIN])
                            && <IconButton className={classes.editButton} size="medium"
                                           onClick={() => setOpen(true)}>
                                <Edit/>
                            </IconButton>}/>
            <Divider/>
            <CardContent>
                <ProductDetails product={product}/>
            </CardContent>
            {hasRole([Role.MANAGER, Role.ADMIN]) &&
                <Fragment>
                    <Divider/>
                    <CardActions>
                        <ProductProfitText profit={formulas.getROI(product.salePrice, product.providerPrice)} isROI/>
                    </CardActions>
                </Fragment>}
            {open && <EditProduct product={product} onClose={() => setOpen(false)}/>}
        </Card>
    );

};

ProductResult.propTypes = {
    product: PropTypes.object
};

export default ProductResult;
