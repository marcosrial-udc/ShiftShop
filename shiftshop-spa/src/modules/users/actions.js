import * as actionTypes from './actionTypes';

export const login = (username, password) => ({
    type: actionTypes.LOGIN_COMPLETED,
    authenticatedUser: ({user: {username, password}})
});

export const logout = () => {
    return {type: actionTypes.LOGOUT};
};
