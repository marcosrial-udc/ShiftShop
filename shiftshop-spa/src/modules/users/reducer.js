import {combineReducers} from 'redux';

import * as actionTypes from './actionTypes';

const initialState = {
    roles: [],
    user: null,
    usersSearch: null,
    blockedUsersSearch: null
};

const roles = (state = initialState.roles, action) => {

    switch (action.type) {

        case actionTypes.GET_ROLES_COMPLETED:
            return action.roles;

        default:
            return state;
    }

};

const user = (state = initialState.user, action) => {

    switch (action.type) {

        case actionTypes.LOGIN_COMPLETED:
            return action.authenticatedUser.user;

        case actionTypes.LOGOUT:
            return initialState.user;

        default:
            return state;
    }

};

const usersSearch = (state = initialState.usersSearch, action) => {

    switch (action.type) {

        case actionTypes.GET_USERS_COMPLETED:
            return action.usersSearch;

        default:
            return state;
    }

};

const blockedUsersSearch = (state = initialState.blockedUsersSearch, action) => {

    switch (action.type) {

        case actionTypes.GET_BLOCKED_USERS_COMPLETED:
            return action.blockedUsersSearch;

        case actionTypes.CLEAR_BLOCKED_USERS_SEARCH:
            return initialState.blockedUsersSearch;

        default:
            return state;
    }

};


const reducer = combineReducers({
    roles,
    user,
    usersSearch,
    blockedUsersSearch
});

export default reducer;
