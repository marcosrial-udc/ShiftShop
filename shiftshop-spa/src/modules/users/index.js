import * as actions from './actions';
import * as actionTypes from './actionTypes';
import reducer from './reducer';
import * as selectors from './selectors';

export {default as Role} from './constants/Role';
export {default as LoginPortal} from './components/LoginPortal';
export {default as LogOut} from './components/LogOut';
export {default as ProfileMenu} from './components/ProfileMenu';

export default {actions, actionTypes, reducer, selectors};
