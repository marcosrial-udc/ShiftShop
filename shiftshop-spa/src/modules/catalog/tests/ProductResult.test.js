import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import {Provider} from 'react-redux';
import {BrowserRouter} from 'react-router-dom';
import {IntlProvider} from 'react-intl';
import {MuiThemeProvider} from '@material-ui/core';
import {SnackbarProvider} from 'notistack';

import {initReactIntl} from "../../../i18n";
import theme from '../../../theme';

import ProductResult from '../components/ProductResult';
import {Role} from '../../users';

jest.mock('../components/ProductDetails', () => () => (<mockProductDetails/>));
jest.mock('../components/ProductProfitText', () => () => (<mockProductProfitText/>));
jest.mock('../components/ProductChangeState', () => () => (<mockProductChangeState/>));

const {messages} = initReactIntl();

describe('ProductResult snapshot', () => {

    const middlewares = [];
    const mockStore = configureStore(middlewares);

    const salesmanUser = {users: {user: {username: 'user', roles: [Role.SALESMAN]}}};
    const adminUser = {users: {user: {username: 'user', roles: [Role.ADMIN]}}};

    const createProductResult = (store, product) => {

        return renderer.create(
            <MuiThemeProvider theme={theme}>
                <SnackbarProvider>
                    <Provider store={store}>
                        <IntlProvider locale='en' messages={messages}>
                            <BrowserRouter>
                                <ProductResult product={product}/>
                            </BrowserRouter>
                        </IntlProvider>
                    </Provider>
                </SnackbarProvider>
            </MuiThemeProvider>
        );

    };

    test('Product null', () => {

        // Initialize mockstore
        const initialState = salesmanUser;
        const store = mockStore(initialState);

        const productResult = createProductResult(store, null);

        expect(productResult.toJSON()).toMatchSnapshot();

    });

    test('Product not null with edit permissions', () => {

        // Initialize mockstore
        const initialState = adminUser;
        const store = mockStore(initialState);

        const product = {};

        const productResult = createProductResult(store, product);

        expect(productResult.toJSON()).toMatchSnapshot();

    });

    test('Product not null without edit permissions', () => {

        // Initialize mockstore
        const initialState = salesmanUser;
        const store = mockStore(initialState);

        const product = {};

        const productResult = createProductResult(store, product);

        expect(productResult.toJSON()).toMatchSnapshot();

    });

});
