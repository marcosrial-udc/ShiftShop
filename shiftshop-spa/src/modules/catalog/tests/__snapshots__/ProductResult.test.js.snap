// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductResult snapshot Product not null with edit permissions 1`] = `
<div
  className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiCardHeader-root"
  >
    <div
      className="MuiCardHeader-content"
    >
      <span
        className="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
      >
        Data
      </span>
    </div>
    <div
      className="MuiCardHeader-action"
    >
      <button
        className="MuiButtonBase-root MuiIconButton-root makeStyles-editButton-5"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
  <hr
    className="MuiDivider-root"
  />
  <div
    className="MuiCardContent-root"
  >
    <mockProductDetails />
  </div>
  <hr
    className="MuiDivider-root"
  />
  <div
    className="MuiCardActions-root MuiCardActions-spacing"
  >
    <mockProductProfitText />
    <div
      className="MuiBox-root MuiBox-root-131"
    >
      <mockProductChangeState />
    </div>
  </div>
</div>
`;

exports[`ProductResult snapshot Product not null without edit permissions 1`] = `
<div
  className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiCardHeader-root"
  >
    <div
      className="MuiCardHeader-content"
    >
      <span
        className="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
      >
        Data
      </span>
    </div>
  </div>
  <hr
    className="MuiDivider-root"
  />
  <div
    className="MuiCardContent-root"
  >
    <mockProductDetails />
  </div>
</div>
`;

exports[`ProductResult snapshot Product null 1`] = `
<div
  className="makeStyles-emptyPlaceholder-1"
>
  <img
    alt="No product"
    className="makeStyles-image-2"
    src="notFound.svg"
  />
  <h6
    className="MuiTypography-root makeStyles-emptyText-3 MuiTypography-subtitle1"
  >
    Product not found
  </h6>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-backToProducts-4"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Back to products page
    </span>
  </button>
</div>
`;
